pkg/appgw/requestroutingrules.go

```diff
@@ -365,15 +365,35 @@ func (c *appGwConfigBuilder) mergePathMap(existingPathMap *n.ApplicationGatewayU
 		existingPathMap.DefaultBackendAddressPool = nil
 		existingPathMap.DefaultBackendHTTPSettings = nil
 	}
-	if pathMapToMerge.PathRules == nil || len(*pathMapToMerge.PathRules) == 0 {
-		return existingPathMap
-	}
 
+	var mergedPathRules, allPathRules []n.ApplicationGatewayPathRule
 	if existingPathMap.PathRules == nil {
-		existingPathMap.PathRules = pathMapToMerge.PathRules
+		allPathRules = *pathMapToMerge.PathRules
 	} else {
-		pathRules := append(*existingPathMap.PathRules, *pathMapToMerge.PathRules...)
-		existingPathMap.PathRules = &pathRules
+		allPathRules = append(*existingPathMap.PathRules, *pathMapToMerge.PathRules...)
+	}
+
+	// we want to ensure that there are only unique paths in the url path map
+	pathMap := make(map[string]n.ApplicationGatewayPathRule)
+	for _, pathRule := range allPathRules {
+		addRuleToMergeList := true
+		for _, path := range *pathRule.Paths {
+			_, exists := pathMap[path]
+			if exists {
+				glog.Errorf("A path-rule with path '%s' already exists in config for BackendPool '%s'. Duplicate path-rule with BackendPool '%s' will not be applied.", path, *pathMap[path].BackendAddressPool.ID, *pathRule.BackendAddressPool.ID)
+				addRuleToMergeList = false
+			} else {
+				pathMap[path] = pathRule
+			}
+		}
+		if addRuleToMergeList {
+			mergedPathRules = append(mergedPathRules, pathRule)
+		}
 	}
+
+	if len(mergedPathRules) > 0 {
```

Comment:
I doubt that this condition will always be satisfied in your code...

Reply:
Not when there are absolutely no path rules. Like in test case requestroutingrule_test -> "test waf policy is not configured in rule path",
it is because of this test case that this condition came into play

Reply:
> Not when there are absolutely no path rules. Like in test case requestroutingrule_test -> "test waf policy is not configured in rule path",
> it is because of this test case that this condition came into play
I think allPathRules in your code will always contain at least one rule...which makes mergedPathRules include at least one path rule
when is all

Reply:
allPathRules is created from input pathMapToMerge.PathRules, and it can be of  length also